### ISSUE TITLE
Buffer type: only appears as `never`

### ISSUE THREAD

**Issue by revmischa** [sensedeep/dynamodb-onetable#302]
**ISSUES will not receive priority that do not follow this template or cannot be easily reproduced or diagnosed**

**Describe the bug**
Trying to use a binary model. 
```ts
models: {
  User {
    binaryData: { type: Buffer, required: true },
  }
}
```

If I do
```ts
User.create({
      binaryData: myBuffer,
    });
```
I get the error:
`Type 'Buffer' is not assignable to type 'undefined'.`

The rest of typings work for me. Just not `Buffer`.

**To Reproduce**

1. Clone this repository. Must test against the current master branch in the repository.
2. Edit your reproducible, MINIMAL, stand-alone test case into the test/debug.ts file.
3. Run via `jest debug` or run VS code in the top level directory.

If you don't edit debug.ts and provide snippets of code, then your issue may not receive priority.

Ensure that your debug.ts includes:

1. Your OneTable schema (complete schema with indexes and models).
2. Actual API code that is failing. Don't use code snippets. Add comments where helpful.
3. Run API with params of {log: true} and include generated DynamoDB API calls
4. The test case is a minimal as possible. Remove all unnecessary code and options.

Check that:

1. You are using a current version of Node and NPM

**Cut/Paste**

Include the code from your debug.ts here and replace the CODE segment. Remember to preserve the three backticks above and below the code block.  See [Formatting Code](https://www.freecodecamp.org/news/how-to-format-code-in-markdown/) for details.

```
PUT YOUR DEBUG.TS CODE HERE.
```

```
PUT YOUR LOG OUTPUT HERE showing the actual DynamoDB command issued.
```

**Expected behavior**

**Screenshots**

**Environment (please complete the following information):**
 - OS
 - Node Version
 - OneTable Version
 - TypeScript Version
 - Any other relevant environment information

**Additional context**

**Comment by NevRA** [comment]
Hey. Any updates on this?

**Comment by mobsense** [comment]
That was fixed in the repo at:

```
diff --git a/src/Model.d.ts b/src/Model.d.ts
index 0b2a46c..3d8a7b6 100644
--- a/src/Model.d.ts
+++ b/src/Model.d.ts
@@ -16,7 +16,7 @@ export type OneType =
     ObjectConstructor |
     StringConstructor |
     SetConstructor |
-    Buffer |
+    BufferConstructor |
     string;
```

**Comment by NevRA** [comment]
@mobsense yep, it will fix schema definition. But when you will try to use it will fail with the mentioned error.
To reproduce: 
1. Clone the latest dynamodb-onetable repo
2. Change strict param to true in samples/typescript/tsconfig.json, without it no reason to specify types all of them would be ANY
3. Change type of models - Account - name to Buffer in samples/typescript/src/schema.js 
4. Check error in samples/typescript/src/index.ts for 
```let account = await Account.create({name: 'Acme Rockets'}) ```

**Comment by mobsense** [comment]
Can you please create a test debug.ts that does this. Then we can quickly fix.

**Comment by NevRA** [comment]
@mobsense debug.ts:
```
/*
    debug.ts - Just for debug

    Edit your test case here and invoke via: "jest debug"

    Or run VS Code in the top level directory and just run.
 */
import {AWS, Client, Entity, Match, Model, Table, print, dump, delay} from './utils/init'
import { OneSchema } from '../src/index.js'

jest.setTimeout(7200 * 1000)

//  Change with your schema
const schema = {
    version: '0.0.1',
    indexes: {
        primary: { hash: 'pk', sort: 'sk' },
        gs1: { hash: 'gs1pk', sort: 'gs1sk', project: 'all' },
    },
    models: {
        User: {
            pk:          { type: String, value: '${_type}#' },
            sk:          { type: String, value: '${_type}#${id}' },

            gs1pk:       { type: String, value: '${_type}#' },
            gs1sk:       { type: String, value: '${_type}#${id}' },

            name:        { type: String },
            email:       { type: String },
            id:          { type: String, generate: 'ulid' },
            buffer:      { type: Buffer },
        }
    }
}

export type AccountType = Entity<typeof schema.models.User>

//  Change your table params as required
const table = new Table({
    name: 'DebugTable',
    client: Client,
    schema,
    logger: true,
})

//  This will create a local table
test('Create Table', async() => {
    if (!(await table.exists())) {
        await table.createTable()
        expect(await table.exists()).toBe(true)
    }
})

test('Test', async() => {
    let User = table.getModel<AccountType>('User')
    let users = await User.create({
        buffer: Buffer.from('test') // TYPE ERROR HERE: Type 'Buffer' is not assignable to type 'never'.
    })
})

test('Destroy Table', async() => {
    await table.deleteTable('DeleteTableForever')
    expect(await table.exists()).toBe(false)
})
```

**Comment by mobsense** [comment]
Thank you. We're on it.

**Comment by NevRA** [comment]
Please note that I removed the type from the schema, this was done on purpose because without this all types for the User model will be any

**Comment by mobsense** [comment]
Thanks, that helped.

The fix is to add: BufferConstructor to EntityFieldFromType in Model.d.ts

```
type EntityFieldFromType<T extends OneField> =
      T['type'] extends (ArrayConstructor | 'array') ? any[]
    : T['type'] extends (BooleanConstructor | 'boolean') ? boolean
    : T['type'] extends (NumberConstructor | 'number') ? number
    : T['type'] extends (ObjectConstructor | 'object') ? object
    : T['type'] extends (DateConstructor | 'date') ? Date
    : T['type'] extends (BufferConstructor) ? Buffer
```

**Comment by mobsense** [comment]
Fixed in 2.3.5

**Comment by NevRA** [comment]
@mobsense works perfectly, thanks!
